site/pages/dp-auctions/index.js

```diff
@@ -0,0 +1,152 @@
+import { DateTime } from 'luxon'
+import Link from 'next/link'
+import orderBy from 'lodash.orderby'
+import useSWR from 'swr'
+import { useState } from 'react'
+import useTranslation from 'next-translate/useTranslation'
+
+import Layout from '../../components/Layout'
+import TokenAmount from '../../components/TokenAmount'
+import fetchJson from '../../utils/fetch-json'
+import ssDpa from '../../utils/dp-auctions'
+import { useUpdatingState } from '../../hooks/useUpdatingState'
+
+const ETH_BLOCK_TIME = 13 // Average block time in Ethereum
+
+// This component renders an auction row. When clicked, the view changes to the
+// auction details page.
+//
+// The remaining time is shown in minutes. So every 10 seconds the remaining
+// time is re-calculated.
+const DPAuctionsRow = function ({ auction }) {
+  const { lang, t } = useTranslation('common')
+
+  const calcEndTime = () =>
+    Date.now() +
+    (auction.endBlock - auction.currentBlock) * ETH_BLOCK_TIME * 1000
+  const endTime = useUpdatingState(calcEndTime, 10000, [auction.currentBlock]) // 10s
+
+  return (
+    <Link href={`dp-auctions/auctions/${auction.id}`}>
+      <tr className="cursor-pointer">
+        <td className="border-2">{auction.id}</td>
+        <td className="border-2">
+          {auction.tokens.map((token) => (
+            <div key={token.address}>
+              <TokenAmount {...token} />
+            </div>
+          ))}
+        </td>
+        <td className="border-2">
+          {auction.status === 'won' || auction.status === 'stopped' ? (
+            '-'
+          ) : (
+            <TokenAmount
+              amount={auction.currentPrice}
+              {...auction.paymentToken}
+            />
+          )}
+        </td>
+        <td className="border-2">
+          {t(auction.status)}
+          {auction.status === 'running' &&
+            ` (${t('ends')} ${DateTime.fromMillis(endTime).toRelative({
+              locale: lang
+            })})`}
+        </td>
+      </tr>
+    </Link>
+  )
+}
+
+// This component renders the list of auctions.
+const DPAuctionsTable = function ({ auctions }) {
+  const { t } = useTranslation('common')
+  const [showEnded, setShowEnded] = useState(false)
+
+  const sortedAuctions = orderBy(
+    auctions,
+    ['endBlock', 'id'],
+    ['desc', 'desc']
+  ).filter((auction) => showEnded || !auction.stopped)
```

Comment:
A minor detail, but if we expect a large number of auctions, perhaps it is better to filter first, and then sort

Reply:
Done in 5c743b4.